### ISSUE TITLE
Unable to clear metadata fault

### ISSUE THREAD

**Issue by GICodeWarrior** [zfsonlinux/zfs#4360]
I have a RAIDZ1 that had two drives fail.  After recovering what I could, the metadata for a directory was still missing.

I still want to try to recover the data in that directory, but for now I need to move forward.  I have images of the raw disks, and I want to clear the metadata error on the recovered pool so I can send a snapshot to a new pool.

Right now, if I try to delete the affected directory, I get an "Invalid Exchange" error (EBADE/ECKSUM).

Is it not possible to clear this fault?  Will I have to cp/mv the unaffected data to a new dataset in order to get away from this error?

(Side note: if anyone has suggestions on how to access the data in the affected directory, that'd be helpful too.)

**Comment by apetc** [comment]
Adding some info on a potentially related issue.

I had a RAIDZ2 (grand total of 7 disks) that had two disks with issues. One disappeared and the other had significant checksum/corruption issues. As the data existed elsewhere and the risk of data loss was not a major concern, I replaced both bad disks and issued appropriate "zpool replace" commands. This kicked off a lengthy resilvering that produced a substantial number of errors (not entirely unexpected).

Once complete, I then ran a scrub to see if it would do any additional cleanup, followed by a clear.

The pool is no longer degraded and shows as ONLINE. It does indicate a number of errors that I can check by running 'zpool status -v'. An attempt to overwrite the damaged files or directories, or even delete them, results in an Invalid Exchange error.

Exact output from an ls in one affected directory: "ls: reading directory .: Invalid exchange"

I'm not sure how to proceed with any further repair or even just remove the affected directories/files and restore from another source.

It almost seems the best option may be to destroy the pool, recreate it, and restore the files from other sources.
